pydata_sphinx_theme/__init__.py

```diff
@@ -38,24 +41,28 @@ def generate_nav_html(kind, startdepth=None, **kwargs):
         HTML string (if kind in ["navbar", "sidebar"])
         or BeautifulSoup object (if kind == "raw")
         """
-        toc_sphinx = context["toctree"](**kwargs)
-        soup = bs(toc_sphinx, "html.parser")
-
         if startdepth is None:
             startdepth = 1 if kind == "sidebar" else 0
 
-        # select the "active" subset of the navigation tree for the sidebar
-        if startdepth > 0:
-            selector = " ".join(
-                [
-                    "li.current.toctree-l{} ul".format(i)
-                    for i in range(1, startdepth + 1)
-                ]
-            )
-            subset = soup.select(selector)
-            if not subset:
-                return ""
-            soup = bs(str(subset[0]), "html.parser")
+        if startdepth == 0:
+            toc_sphinx = context["toctree"](**kwargs)
+        else:
+            toc_sphinx = index_toctree(app, pagename, startdepth, **kwargs)
+
+        soup = bs(toc_sphinx, "html.parser")
+
+        # # select the "active" subset of the navigation tree for the sidebar
```

Comment:
just making sure you notice this is still here :-)

Reply:
Thanks for the reminder ;) I still need to clean that up. 

I am still a little bit wondering if my new code is fully equivalent in all cases to this beautifulsoup selector, but we will probably need to try out in practice to be sure ;) 
(I checked locally with pandas and geopandas docs, and that seems all good)